Providers/Scripts/2.4x-2.5x/Scripts/Tests/run_test_OMS_nxProviders.py

```diff
@@ -0,0 +1,153 @@
+os.system('cp /etc/sudoers /etc/sudoers.back')
+os.system('cat /etc/opt/microsoft/omsagent/sysconf/sudoers >> /etc/sudoers')
+os.system('chmod 440 /etc/sudoers')
+# Setup python __init__.py.
+os.system('su omsagent -c "/opt/microsoft/omsconfig/Scripts/RegenerateInitFiles.py"')
+# Import the keys
+os.system('su omsagent -c "/opt/microsoft/omsconfig/Scripts/ImportGPGKey.sh /opt/microsoft/omsconfig/keys/msgpgkey.asc keymgmtring.gpg"')
+os.system('su omsagent -c "/opt/microsoft/omsconfig/Scripts/ImportGPGKey.sh /opt/microsoft/omsconfig/keys/dscgpgkey.asc keyring.gpg"')
+#os.system('mkdir -p /var/opt/microsoft/omsconfig')
+#os.system('chgrp -R omsagent /var/opt/microsoft/omsconfig')
+
+# Setup fake syslog/rsyslog oms conf files.
+if not os.path.exists('/etc/opt/omi/conf/omsconfig/syslog-ng-oms.conf'):
+    os.system('touch /etc/opt/omi/conf/omsconfig/syslog-ng-oms.conf')
+if not os.path.exists('/etc/opt/omi/conf/omsconfig/rsyslog-oms.conf'):
+    os.system('touch /etc/opt/omi/conf/omsconfig/rsyslog-oms.conf')
+
+if os.path.exists('/etc/rsyslog.d/'):
+    os.system('cp /etc/opt/omi/conf/omsconfig/rsyslog-oms.conf /etc/rsyslog.d/95-omsagent.conf')
+    os.system('chown omsagent:omsagent /etc/rsyslog.d/95-omsagent.conf')
+os.system('mkdir -p /etc/opt/microsoft/omsagent/conf/')
+os.system('touch /etc/opt/microsoft/omsagent/conf/omsagent.conf')
+os.system('chown -R omsagent /etc/opt/microsoft/omsagent/conf/')
+os.system('mkdir -p /opt/microsoft/omsagent/bin')
+os.system('/bin/echo -e "#!/bin/bash\necho \n" > /opt/microsoft/omsagent/bin/service_control')
+os.system('chmod +x /opt/microsoft/omsagent/bin/service_control')
+'/opt/microsoft/omsagent/bin/service_control'
+os.system('chown -R omsagent /var/opt/microsoft/omsconfig')
+os.system('chmod +w /etc/opt/omi/conf/omsconfig/*')
+os.system('chown -R omsagent /etc/opt/omi/conf/omsconfig')
+
+# Copy the Tests
+destpath = '/opt/microsoft/omsconfig/Scripts/' + sys.argv[2] + '/Scripts/'
+srcpath = '/' + os.path.join(*os.path.realpath(__file__).split('/')[:-1])
+os.system('cp -r ' + srcpath + ' ' + destpath)
+os.system('chown -R omsagent /opt/microsoft/omsconfig/Scripts')
+#os.system('chgrp -R omsagent /opt/microsoft/omsconfig/Scripts')
+os.chdir(destpath + 'Tests')
+
+#Don't restart the service as it is not installed.
+os.system("echo '#!/bin/bash' > /opt/microsoft/omsagent/bin/service_control")
+
+#nxUser
+os.system('/usr/sbin/useradd -d "/home/jojoma" -m  -g mail jojoma')
+os.system('/usr/sbin/usermod -c "JO JO MA,JOJOMA" jojoma')
+
+#nxGroup
+os.system('/usr/sbin/groupadd -g 1101 jojomamas')
+
+#nxPackage
+if os.system('which dpkg 2> /dev/null') == 0:
+    os.system('/usr/bin/dpkg -i ' + destpath + 'Tests/dummy-1.0.deb')
+else:
+    os.system('/bin/rpm -i ' + destpath + 'Tests/dummy-1.0-1.x86_64.rpm')
+# Call the test!
+result = os.system('su -c " python ' + destpath + 'Tests/test_OMS_nxProviders.py" omsagent')
```

Comment:
You are correct.  Right now this is designed to run on a build system during the make.
I'm not sure that any build systems have multiple versions of python installed.
The different version on the same box is a feature TBD.  Once I've handed this off to your OMS team, you'll be able to modify it's logic in any way.